tests/tracer/test_tracer.py

```diff
@@ -1702,3 +1703,37 @@ def process_trace(self, trace):
     # Force gc
     gc.collect()
     assert len(spans) == 0
+
+
+def test_processor_shutdown():
+    class Processor(SpanProcessor):
+        def __init__(self):
+            super(Processor, self).__init__()
+            self.shutdown_called = 0
+
+        def on_span_start(self, span):
+            pass
+
+        on_span_finish = on_span_start
+
+        def shutdown(self):
+            # type: () -> None
+            self.shutdown_called += 1
+
+    # tracer.shutdown()
+    t = Tracer()
+    p = Processor()
+    t._add_span_processor(p)
+    assert p.shutdown_called == 0
+    t.shutdown()
+    assert p.shutdown_called == 1
+
+    # tracer.configure()
+    t = Tracer()
+    p = Processor()
+    t._add_span_processor(p)
+    assert p.shutdown_called == 0
+    t.configure()
+    assert p.shutdown_called == 1
+    t.shutdown()
+    assert p.shutdown_called == 1
```

Comment:
yeah, that is what is happening, it isn't clear from this test that `p` gets removed from the list of span processors.

Reply:
Yeah I'm not really a fan of `configure` having this behaviour... maybe it shouldn't?

I guess "the way" to add span processors to the current api would be with `configure(span_processors=...)` and then don't shutdown/initialize them unless a value is passed.

Reply:
Yep as you @Kyle-Verhoog point out span processors needs to be exposed in the public api somehow which right now is limited to just `configure`.

I wanted to back up a little bit and ask a simpler question: why do span processors have to be re-initialized at the moment if they never change? Is it because the writer can be recreated in `configure`?

If that's the case, sharing the writer between the tracer and the span processors might be worth reconsidering.

Reply:
we don't expose them at all, it is "trace processors" (sort of) that get exposed via filters right?

Reply:
I think I am ok with this test, but 100% please add a comment on why it is `1`, it isn't obvious from looking at this code why the behavior is like this

Reply:
I'm going to simplify this PR to only introduce the shutdown method and then follow up with the logic of using it in `Tracer`